Source/DafnyServer/CounterExampleGeneration/DafnyModel.cs

```diff
@@ -154,6 +160,64 @@ public class DafnyModel {
       return result;
     }
 
+    /// <summary>
+    /// Registered all values of basic types specified by the model in
+    /// the reservedValues map;
+    /// </summary>
+    private void RegisterReservedValues() {
+      var fCharToInt = Model.MkFunc("char#ToInt", 1);
+      reservedValues[Type.Char] = new();
+      foreach (var app in fCharToInt.Apps) {
+        var UTFCode = int.Parse(((Model.Integer)app.Result).Numeral);
+        reservedValues[Type.Char].Add(Convert.ToChar(UTFCode).ToString());
+      }
+
+      var fU2Int = Model.MkFunc("U_2_int", 1);
+      reservedValues[Type.Int] = new();
+      foreach (var app in fU2Int.Apps) {
+        // this skips negative values
+        if (app.Result is Model.Integer) {
```

Comment:
C#'s pattern matching allows declaring a variable inline, making the cast on the following redundant, for example:
```cs
if (app.Result is Model.Integer integerResult) {
  reservedValues[Type.Int].Add(integerResult.Numeral);
}
```

Reply:
Thank you for catching this! I have made the change.